zap/src/main/java/org/parosproxy/paros/control/Control.java

```diff
@@ -330,6 +332,29 @@ public void run() {
         }
     }
 
+    /**
+     * Sets the non zero value ZAP will exit cleanly with. ZAP may still exit with a non zero value
+     * if a serious error occurs. This will work however ZAP is run but it makes more sense if ZAP
+     * is run in cmdline mode. Any add-on can set an exit status so attempts to reset the status to
+     * zero will be rejected.
+     *
+     * @param exitValue the non zero value ZAP will exit it with
+     * @param logMessage the message that will be logged at info level
+     * @since 2.11.0
+     */
+    public void setExitStatus(int exitStatus, String logMessage) {
+        if (exitStatus == 0) {
+            log.error("Not setting the exit status to zero - culprit: {}", logMessage);
```

Comment:
`Now`?

The condition is equal zero so ??

Reply:
Its an error message saying that we're _not_ going to set the status to zero, as thats the default and something else might have already set it to something non zero and we dont want to lose that.

Reply:
But we're checking that what's being passed is zero, so zero is being set from outside (non-default) but still zero?

Actually I guess it's kind of ambiguous, how does java handle that? Local scope first? Since both the passed var and class var are the same name.

Reply:
It will test the parameter, ie prevent add-ons from resetting the exit status to zero. In order to test the class var you'd need to use this.exitStatus

Reply:
Ok, I don't see why the caller can't or shouldn't set zero, but fair enough.

Reply:
It's to prevent add-ons from overriding the failure status set by others (0 is the default anyway).